File: lib/Backend/server/models/ngo_model.js
const mongoose = require("mongoose");
const ratingSchema = require('./rating');
const ngosSchema = mongoose.Schema({
  ngo_id: {
    required: true,
    type: String,
    trim: true,
  },
  name: {
    required: true,
    type: String,
    trim: true,
  },
  address: {
    type: String,
    trim: true,
    default: "",
  },
  password: {
    type: String,
    trim: true,
    required: true,
   
  },
  phone_number: {
    type: String,
    required: true,
    trim: true,
  },
  email1:{
    type: String,
    required: true,
    trim: true,
  },
  email: {
    type: String,
    trim: true,
    required: true,
    //for validation like @, .com
    // validate: {
    //   validator: (value) => {
    //     const re =
    //       /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
    //     return value.match(re);
    //   },
    //   message: "Please enter a valid  address",
    // },
  },
  ratings: [ratingSchema]
});
const NGO = mongoose.model("NGO", ngosSchema);
module.exports = NGO;

File: lib/FrontEnd/models/ngo_model.dart
import 'dart:convert';

class NGO {
  final String ngo_id;
  final String name;
  final String address;
  final String password;
  final String phone_number;
  final String email1;
  final String email;
  final String token;
  //final List<Rating>? rating;
  NGO({
    //required this.address,
    required this.ngo_id,
    required this.name,
    required this.address,
    required this.password,
    required this.phone_number,
    required this.email1,
    required this.email,
    required this.token,
    //this.rating,
  });
  Map<String, dynamic> toMap() {
    final result = <String, dynamic>{};

    result.addAll({'ngo_id': ngo_id}); // include the _id field here
    result.addAll({'name': name});
    result.addAll({'address': address});
    result.addAll({'password': password});
    result.addAll({'phone_number': phone_number});
    result.addAll({'email1': email1});
    result.addAll({'email': email});
    result.addAll({'token': token});
    //result.addAll({'rating': rating});

    return result;
  }

  factory NGO.fromMap(Map<String, dynamic> map) {
    return NGO(
      ngo_id: map['ngo_id'] ?? '',
      name: map['name'] ?? '',
      address: map['address'] ?? '',
      password: map['password'] ?? '',
      phone_number: map['phone_number'] ?? '',
      email1: map['email1'] ?? '',
      email: map['email'] ?? '',
      token: map['token'] ?? '',
      // rating: map['ratings'] != null
      //     ? List<Rating>.from(
      //         map['ratings']?.map((x) => Rating.fromMap(x)),
      //       )
      //     : null);
    );
  }

  String toJson() => json.encode(toMap());

  factory NGO.fromJson(String source) => NGO.fromMap(json.decode(source));
}
